public/javascripts/create-password.js

```diff
@@ -0,0 +1,75 @@
+const continueEvent = document.querySelector("#continue-button");
+const passwordFields = document.querySelector("#password-fields");
+const formGroup = document.querySelector(".govuk-form-group");
+const passwordInput = document.querySelector("#password");
+const confirmPasswordInput = document.querySelector("#confirm-password");
+const errorTextSection = document.querySelector("#error");
+const errorSummarySection = document.querySelector("#error-summary");
+
+continueEvent.addEventListener("click", checkPassword);
+
+let passwordMinimumLength = 8;
+let passwordRegex = /^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]/;
+
+function addErrorClasses(formGroup, passwordInput, confirmPasswordInput) {
+  formGroup.classList.add("govuk-form-group--error");
+  passwordInput.classList.add("govuk-input--error");
+  confirmPasswordInput.classList.add("govuk-input--error");
+}
+
+function addAriaAttribute(passwordInput, confirmPasswordInput) {
+  passwordInput.setAttribute(
+    "aria-describedby",
+    "password-hint password-error"
+  );
+  confirmPasswordInput.setAttribute(
+    "aria-describedby",
+    "password-hint password-error"
+  );
+}
+
+function showError(errorType) {
+  addErrorClasses(formGroup, passwordInput, confirmPasswordInput);
+  addAriaAttribute(passwordInput, confirmPasswordInput);
+
+  if (errorType === "length") {
+    errorMessage = "Password must be 8 characters or more";
+  } else if (errorType === "characters") {
+    errorMessage =
+      "Enter a password with at least one number and one special character";
+  } else if (errorType === "unmatched") {
+    errorMessage = "Both passwords should match";
+  } else {
+    return;
+  }
+
+  errorTextSection.innerHTML = `<p id="password-error" class="govuk-error-message" data-cy="password-error">
```

Comment:
Could most of this be in the HTML already and then just unhide it and update the `span` with the `errorMessage` variable by giving it an id ?

Reply:
Looked at the page source of a couple of government forms and couldn't see error sections within the HTML. When I tried using no display with css, I could see the HTML in page source. Thought this was a way of hiding as much as possible